models/_sector/dex/trades/dex_trades_beta.sql

```diff
@@ -0,0 +1,31 @@
+{{ config(
+    tags=['dunesql'],
+    schema = 'dex',
+    alias ='trades_beta',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'project', 'version', 'tx_hash', 'evt_index']
+    )
+}}
+
+-- (blockchain, project, project_version, model)
```

Comment:
ideally, we add a fifth element: `project_start_date` and pull from each model as they are in current dex trades

then we can pass into macro, apply else filter on inner join to base tx table

the way the macro is set up now though, it loops through base project models, but then just one join to base tx table, so not feasible. it may also be overkill to do all of this, so just suggesting to look into

Reply:
i don't think we'd want to loop around all tables and join base tx each time, that would be too many joins. so not sure my full thoughts here yet, thinking out loud

Reply:
I actually use the similar design as `map-reduce` for the joining stage. I divided the dex.trades pipelines into:

1. base trades info (directly read/parse from decoded sources table) from each dex.
2. joining with transaction info for a specific chain, which these trades from different dexs shared. (ideally this *.transactions is only reading for once for each chain).
3. merge trades from different chains into full dex.trades table.

Reply:
```
union stage:
[dex1 + dex2 + dex3 + dex4 + ...] on the same chain
join stage 1    || (Reading only once big table *.transactions)
                || Joining *.transactions using (tx_hash, evt_index)
               \\//
[dex.trades on a specific chain]
join stage 2    || (Reading only once big table prices.usd where blockchain = *)
                || Joining a specific chain's prices.usd using (block_time)
               \\//
[detailed dex.trades on a specific chain] + [detailed dex.trades on other chains]
```

Reply:
Also here is another implementation: #4607